src/scroll-timeline-base.js

```diff
@@ -449,45 +449,61 @@ function range(timeline, phase) {
   }
 
   const inset = parseInset(details.inset, containerSize);
+
+
+  // Cover:
+  // 0% progress represents the position at which the start border edge of the
+  // element’s principal box coincides with the end edge of its view progress
+  // visibility range.
+  // 100% progress represents the position at which the end border edge of the
+  // element’s principal box coincides with the start edge of its view progress
+  // visibility range.
   const coverStartOffset = viewPos - containerSize + inset.end;
   const coverEndOffset = viewPos + viewSize - inset.start;
+
+  // Contain:
```

Comment:
Well-defined, and is it also backed by spec?

Reply:
A recent spec updated defined all these: https://drafts.csswg.org/scroll-animations-1/#view-timelines-ranges